pkg/httputils/http_util.go

```diff
@@ -454,3 +469,22 @@ func handlePairRange(rangeStr string, length int64) (*RangeStruct, error) {
 		EndIndex:   endIndex,
 	}, nil
 }
+
+// RegisterProtocol registers custom protocols in global variable "protocols" which will use in dfget and supernode
+// Example:
+//   protocols := "helloworld"
+//   newTransport := funcNewTransport
+//   httputils.RegisterProtocol(protocols, newTransport)
+func RegisterProtocol(scheme string, rt http.RoundTripper) {
+	protocols.Store(scheme, rt)
+}
+
+// RegisterProtocolOnTransport registers all new protocols in "protocols" for a special Transport
+// this function will be used in supernode and dfwget
+func RegisterProtocolOnTransport(tr *http.Transport) {
```

Comment:
Actually I am not on the way to implement register protocol. I have to say we can make it by this indeed. While I am afraid that in your code, maybe I have two kinds of suggestions:
1.  `func (tr  *http.Transport) RegisterProtocol( key string, value http.RoundTripper)`  would be much more direct. We could try to make the pattern more object oriented. If we cannot add a function for the built-in struct, we can design one based on it.
2. If we could do things above, the function would be a fundamental one, and we can use the range loop in caller rather the implementation part.

WDYT?